### DISCUSSION TITLE
The example HelloWorldExampleDataSharing, why the is_bounded() return false?

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3710] [General]
#ifdef TOPIC_DATA_TYPE_API_HAS_IS_BOUNDED
    eProsima_user_DllExport inline bool is_bounded() const override
    {
        return false;
    }

As the Constraints, the is_bounded()  must return true, right?

**Comment by JesusPoderoso** [comment]
Hi @wjhtinger , nice catch!
You are completely right. In that example we are [forcing the publisher to enable data-sharing](https://github.com/eProsima/Fast-DDS/blob/c92d2f95b78141f16048c920e2979d0c39409865/examples/cpp/dds/HelloWorldExampleDataSharing/HelloWorldPublisher.cpp#L77), but the IDL used to generate the example types defines the message as `string message`, which is unbounded, instead of `char message[20]`. 
I've opened a pull request to fix it #3714, thanks!
